### ISSUE TITLE
[doc] Comparison to ApplySyntax

### ISSUE THREAD

**Issue by eugenesvk** [jfcherng-sublime/ST-AutoSetSyntax#8]
Any chance you could add a brief comparison of AutoSetSyntax and [ApplySyntax](https://github.com/facelessuser/ApplySyntax) to the docs to help users make an informed choice about your great plugins?
Thank you!

**Comment by jfcherng** [comment]
tl;dr: I would say AutoSetSyntax is more generic and powerful but its configuration is cumbersome. 

---

The major problem is that I have never used ApplySyntax (since the very first commit of AutoSetSyntax is before ApplySyntax's), so I can't tell precisely what's the difference. But I did read its doc when I created v2. From what I read, I think

### Pros

- Infinite level nested rule definition (usually, one level is enough though)
- Machine learning based syntax guessing (i.e., `guesslang`. VSCode has it default enabled so I think it's mature enough for production but I still have it default disabled as a opt-in)
- Configuration autocompletion with  [LSP-json](https://packagecontrol.io/packages/LSP-json)
- Uses Python 3.8 along with fully type-annotated source codes.

### Cons

- The configuration is cumbersome.

Others are more of style differences and should be convert-able between these two plugins.

Such as custom plugin: [AutoSetSyntax = class](https://jfcherng-sublime.github.io/ST-AutoSetSyntax/advanced-topics/custom-constraint/) and [ApplySyntax = function](http://facelessuser.github.io/ApplySyntax/usage/#function-rule)

**Comment by eugenesvk** [comment]
> have never used ApplySyntax

Oh, ok, I just assumed otherwise since you mentioned it as an inspiration

**Comment by jfcherng** [comment]
> Oh, ok, I just assumed otherwise since you mentioned it as an inspiration

Inspiration from docs :D

**Comment by eugenesvk** [comment]
By the way, one thing I've noticed that your plugin does that ApplySyntax doesn't — yours includes `extension` in the set of rules, so you `match "all"` filter applies to it, while in ApplySyntax it's not part of the rules, so a `first_line` rule will match any extension regardless of the `extension` syntax setting
(also there is a warm glow from using the newer 3.8 plugin host)
